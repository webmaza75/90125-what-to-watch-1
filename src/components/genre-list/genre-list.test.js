import React from 'react';
import renderer from 'react-test-renderer';

import GenreList from './genre-list.jsx';

const filter = `Drama`;
const genres = [`All genres`, `Comedy`, `Drama`, `Triller`];

it(`GenreList correctly renders`, () => {
  const genreList = renderer
    .create(<GenreList
      genres={genres}
      activeFilter={filter}
      onGenreChange={jest.fn()}
    />)
    .toJSON();

  expect(genreList).toMatchSnapshot();
});
