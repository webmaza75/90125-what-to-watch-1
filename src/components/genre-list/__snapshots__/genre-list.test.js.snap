// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList correctly renders 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      Triller
    </a>
  </li>
</ul>
`;
